### PR TITLE
Player manager

### LINKED ISSUE #20: Player manager
**Description**
Implement a player manager singleton that keeps track of all the players in the game.

### PR DESCRIPTION
**Pull request by smeas**
Closes #20 

**Description**  
The player manager is a singleton that keeps track of all the player in the game. It also has the PlayerInputManager and controls joining of new players.

**List of changes**  
- Added player manager and prefab

### CODE REVIEW

**Review by MrJackey — COMMENTED** [review]
Nothing breaking, just a few questions. Great work 👍

File: Assets/00_Content/Scripts/Player/PlayerManager.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+		public event Action<PlayerComponent> PlayerLeft;
+
+		protected override void Awake() {
+			base.Awake();
+
+			playerInputManager = GetComponent<PlayerInputManager>();
+			playerInputManager.onPlayerJoined += OnPlayerJoined;
+			playerInputManager.onPlayerLeft += OnPlayerLeft;
+
+			DontDestroyOnLoad(this);
```
[the remark this answers is not in the document]
Comment (smeas) [reply Assets/00_Content/Scripts/Player/PlayerManager.cs:46]: Correct, but it has the same effect. Should maybe have used `gameObject` instead for clarity.

File: Assets/00_Content/Scripts/Player/PlayerManager.cs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+			return null;
+		}
+
+		#region Event Handlers
+
+		private void OnPlayerJoined(PlayerInput playerInput) {
+			PlayerComponent player = playerInput.GetComponent<PlayerComponent>();
+			if (player == null) {
+				Debug.Assert(false, "Joined player has PlayerComponent");
```
[the remark this answers is not in the document]
Comment (smeas) [reply Assets/00_Content/Scripts/Player/PlayerManager.cs:64]: Assertion messages are often written like this.
> I assert that the *joined player has a player component*.

**Review by MrJackey — APPROVED** [review]
:shipit:

**Review by tzaarela — APPROVED** [review]
Snyggt! Inget att anmärka på.

**Pull request merged**